### PR TITLE
Added shimmer effect on price loading

### LINKED ISSUE #2510: Change price loading effects in swap UI
**Description**

Based on the Michel's comment in fast price PR https://github.com/gnosis/cowswap/pull/2477#issuecomment-1054097219, we would like to add something like `shimmer effect` to some swap UI elements while the price is loading.

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

Fixes #2510

This will show shimmer effect and shimmer fields until the best quote is initially loaded and also when the price is being re-fetched.

Screenshot from 2022-03-04 16-59-11

Screenshot from 2022-03-04 17-02-36

### CODE REVIEW

**Comment by biocom** [comment]
Looks great! I think this really makes it more clear which elements are being updated. I think for a next PR we could look at preventing certain nudging/shifting of elements. Probably some elements briefly changing height? Otherwise I think this is a great addition.

File: src/custom/components/Shimmer/Shimmer.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  borderRadius?: string
+  lightColor?: string
+  darkColor?: string
+}
+
+const StyledShimmer = styled(Shimmer)<ShimmerProps>`
+  border-radius: ${({ borderRadius }) => borderRadius || `15px`};
+  background: linear-gradient(
+    to right,
+    ${({ lightColor, theme }) => lightColor || lighten(theme.darkMode ? 0.5 : 0.3, theme.bg1)} 8%,
```
Comment (biocom) [inline src/custom/components/Shimmer/Shimmer.tsx:17]: We could consider adding this gradient in the theme, so we could re-use it.

**Review by anxolin — COMMENTED** [review]
This PR makes the UI look slower for some reason. 

Before we use to show all the numbers as soon as we have the first price, not it looks like is waiting. 

Also, when refreshes the price, is hiding the numbers, so its  bit impractical, because a refresh will liley give you a similar number, so nice to see the number while is refreshing

**Comment by biocom** [comment]
On a second look, I agree with @anxolin, given we already have a price, we could already show this. This as I realize again now, we're not waiting for 1 single price request to return. Which in that case it could make sense to use a fully covering shimmer effect.

Screen Shot 2022-03-07 at 10 14 49

What I'd suggest we should do for now, is resort to using the shimmer effect only on the background (like we had). Also remove it from the footer/price indicator.

On a future iteration we can perhaps explore other (non blocking) ways to display the fact that it's still updating (outside having that shimmer effect in the background). But maybe that turns out not to even be required.

**Comment by W3stside** [comment]
I was under the impression this would ONLY show for long loads and/or initial price load? not for subsequent reloads.

so: 
1. first time app load and fetch -> SHIMMER
2. same token pair as 1), price refresh? -> NO SHIMMER
3. change token pair and fetch new price? -> SHIMMER

**Comment by elena-zh** [comment]
I'd like to add these cases: fee and price impact warnings are displayed when the prices are being updated under the shimmer effect. I think, it would be nice to hide them at this moment
loading - displayed
the same

**Pull request closed without merging**